### DISCUSSION TITLE
Adding "support" for rootless Docker

### DISCUSSION THREAD

**Discussion by macvenez** [louislam/dockge#357] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Hi!
I just tried installing dockge on my host and I noticed the default docker socket is `/var/run/docker.sock` (as it should be on a default root Docker installation).

I installed Docker as rootless so my socket location is different, something like `/run/user/1000/docker.sock` so I had to modify the pregenerated `compose.yaml` file. (Nothing too difficult but I think it would be cool to have this automatically checked)

### ✔️ Solution

I think it may be useful to add the option [here](https://dockge.kuma.pet/) to use a custom Docker socket or a simple checkbox to use the default rootless Docker socket (alternatively reading `DOCKER_HOST` environment variable).

**Comment by smgoller** [comment]
dockge apparently uses the docker cli under the hood, so it respects the DOCKER_HOST environment variable. I modified my dockge compose.yaml to include a socket-proxy service and set DOCKER_HOST appropriately and it works fine. [Here](https://gist.github.com/smgoller/23dcaf430e762e580cde18cd5b874fc5) is a gist of a compose.yaml with dockge and socket-proxy.

**Reply by willyp713** [reply]
Confirmed that this works. Maybe a simple documentation edit is needed so people know it's an option.

**Reply by macvenez** [reply]
I understand the socket proxy thing but wouldn't it be easier to just change the bind mount for the Dockge container?
Thanks

**Reply by qsevers23** [reply]
Sorry to revive an older thread, did you guys have any success with this while trying to authenticate with private container registries? I keep getting 403 unauthorized from gitlab when I mount `/home/<rootless_docker_user>/.docker/config.json` into the container.

**Reply by stereotypy** [reply]
Hi, does anyone have a simpler method just changing bind mount? I was looking here to see examples but couldn't seem to get it to work: https://stackoverflow.com/questions/68331153/docker-rootless-error-you-need-to-share-your-docker-host-socket-with-a-volume-a

**Reply by macvenez** [reply]
> Hi, does anyone have a simpler method just changing bind mount? I was looking here to see examples but couldn't seem to get it to work: https://stackoverflow.com/questions/68331153/docker-rootless-error-you-need-to-share-your-docker-host-socket-with-a-volume-a

On my setup I'm using:
```
volumes:
      - /run/user/1000/docker.sock:/var/run/docker.sock
```
and it works

**Reply by stereotypy** [reply]
Thanks so much, that worked perfectly!
